File: Automotive.Marketplace.Application/Features/ListingFeatures/GetMyListings/GetMyListingsResponse.cs
using Automotive.Marketplace.Application.Models;

namespace Automotive.Marketplace.Application.Features.ListingFeatures.GetMyListings;

public sealed record GetMyListingsResponse
{
    public Guid Id { get; set; }
    public decimal Price { get; set; }
    public int Mileage { get; set; }
    public bool IsUsed { get; set; }
    public string City { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public int Year { get; set; }
    public string MakeName { get; set; } = string.Empty;
    public string ModelName { get; set; } = string.Empty;
    public ImageDto? Thumbnail { get; set; }
    public int ImageCount { get; set; }
    public int DefectCount { get; set; }
}

File: automotive.marketplace.client/src/features/myListings/types/GetMyListingsResponse.ts
type Image = {
  url: string;
  altText: string;
};

export type GetMyListingsResponse = {
  id: string;
  price: number;
  mileage: number;
  isUsed: boolean;
  city: string;
  status: string;
  year: number;
  makeName: string;
  modelName: string;
  thumbnail: Image | null;
  images: Image[];
  imageCount: number;
  defectCount: number;
  fuelName: string;
  transmissionName: string;
  engineSizeMl: number;
  powerKw: number;
  likeCount: number;
  conversationCount: number;
};
